### PR TITLE
 contibution messages, links are recognised

### LINKED ISSUE #2247: 🔧 [Refactor] links in contribution messages clickable
## 🔧 Refactor ticket
make the links in the text of the contribution messages clickable.  a new window should open.

### PR DESCRIPTION
**Pull request by ogerly**
## 🍰 Pullrequest
in the contibution messages, links are recognised, found and placed in an <A> element. with target _blank. 

### Issues
 - fixes #2247  

### adminarea
image

### webapp
image

### CODE REVIEW

**Review by Mogge — COMMENTED** [review]
Please define the regex pattern outside the vue component.
```
const PATTERN = /.../

export defaut {
```

Have something like:
```
  computed: {
      linkifiedMessage() {
        return this.messageObject.message.replace(PATTERN, '...')
      }
  }
```

File: frontend/src/components/ContributionMessages/ContributionMessagesListItem.vue [hunk 1]
```diff
@@ -34,5 +37,17 @@ export default {
...
+  methods: {
+    linkify(inputText) {
+      // console.log(inputText.match(this.pattern))
+      this.messageObject.message = inputText.replace(
+        this.pattern,
+        "<i><a href='$1' target='_blank'>$1</a></i>",
+      )
+    },
+  },
+  created() {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply frontend/src/components/ContributionMessages/ContributionMessagesListItem.vue]: done

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 2]
```diff
@@ -23,5 +23,25 @@ export default {
       },
     },
   },
+  data() {
+    return {
+      pattern:
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue]: done

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 3]
```diff
@@ -23,5 +23,25 @@ export default {
...
+  methods: {
+    linkify(inputText) {
+      // console.log(inputText.match(this.pattern))
+      this.messageObject.message = inputText.replace(
+        this.pattern,
+        "<i><a href='$1' target='_blank'>$1</a></i>",
+      )
+    },
+  },
+  created() {
```
[the remark this answers is not in the document]
Comment (ogerly) [reply admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue]: done

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 4]
```diff
@@ -1,21 +1,34 @@
...
       <span class="ml-2 mr-2">{{ message.userFirstName }} {{ message.userLastName }}</span>
       <span class="ml-2">{{ $d(new Date(message.createdAt), 'short') }}</span>
       <small class="ml-4 text-success">{{ $t('moderator') }}</small>
-      <div class="mt-2">{{ message.message }}</div>
+      <div class="mt-2">
+        <span v-for="({ type, text }, index) in linkifiedMessage" :key="index">
+          <b-link v-if="type === 'link'" :to="text">{{ text }}</b-link>
+          <span v-else>{{ text }}</span>
+        </span>
+      </div>
```
Comment (ulfgebhardt) [inline admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:13]: can we remove this duplicate code and put it outside the `v-if v-else` ?

File: admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue [hunk 5]
```diff
@@ -1,21 +1,34 @@
...
     </div>
   </div>
 </template>
 <script>
+const PATTERN =
```
Comment (ulfgebhardt) [inline admin/src/components/ContributionMessages/slots/ContributionMessagesListItem.vue:29]: Please describe this better so one can search for this:

```suggestion
const LINK_REGEX_PATTERN =
```

```suggestion
const LINK_REGEX =
```

File: frontend/src/components/ContributionMessages/ContributionMessagesListItem.vue [hunk 6]
```diff
@@ -1,22 +1,34 @@
 <template>
```
Comment (ulfgebhardt) [inline frontend/src/components/ContributionMessages/ContributionMessagesListItem.vue:1]: adjust this file accordingly to the other suggestions please.

File: frontend/src/components/Contributions/ContributionForm.vue [hunk 7]
```diff
@@ -88,6 +88,8 @@
   </div>
 </template>
 <script>
+const PATTERN_NON_DIGIT = /\D/g
```
Comment (ulfgebhardt) [inline frontend/src/components/Contributions/ContributionForm.vue:91]: maybe use a standard for regex definitions when it comes to naming - how would that standard look like?

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]

File: frontend/src/components/Contributions/ContributionForm.vue (hunk 7 above)
Comment (Mogge) [reply frontend/src/components/Contributions/ContributionForm.vue:91]: This is bad code. I focused on the links. To validate the amount field, we need a proper solution, better than this regex. But this is another issue.

**Review by ulfgebhardt — DISMISSED** [review]

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**